src/boss/lambdastore/store.go

```diff
@@ -104,14 +99,13 @@ func (s *LambdaStore) DeleteLambda(w http.ResponseWriter, r *http.Request) {
 
 func (s *LambdaStore) ListLambda(w http.ResponseWriter, r *http.Request) {
 	s.mapLock.Lock()
+	defer s.mapLock.Unlock()
```

Comment:
Holding a lock for the whole execution is usually a good idea, but I do try to avoid it when doing I/O, so perhaps the previous version was better.